msp/mspimpl.go

```diff
@@ -837,6 +843,32 @@ func (msp *bccspmsp) IsWellFormed(identity *m.SerializedIdentity) error {
 	if bl.Type != "CERTIFICATE" && bl.Type != "" {
 		return errors.Errorf("pem type is %s, should be 'CERTIFICATE' or missing", bl.Type)
 	}
-	_, err := x509.ParseCertificate(bl.Bytes)
-	return err
+	cert, err := x509.ParseCertificate(bl.Bytes)
+	if err != nil {
+		return err
+	}
+
+	return isIdentitySignedInCanonicalForm(cert.Signature, identity.Mspid, identity.IdBytes)
+
+}
+
+func isIdentitySignedInCanonicalForm(sig []byte, mspID string, pemEncodedIdentity []byte) error {
+	r, s, err := utils.UnmarshalECDSASignature(sig)
+	if err != nil {
+		return err
+	}
+
+	expectedSig, err := utils.MarshalECDSASignature(r, s)
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Equal(expectedSig, sig) {
+		encoding := base64.StdEncoding
+		return errors.Errorf("identity %s for MSP %s has invalid CA signature: expected %s but got %s",
+			string(pemEncodedIdentity), mspID, encoding.EncodeToString(expectedSig),
+			encoding.EncodeToString(sig))
```

Comment:
I'm not sure that logging those bytes is particularly helpful...

Reply:
How so? under normal circumstances we should never ever see this error. 
If we do happen to see it, one can use it to investigate what's going on.

Reply:
Yeah but those bytes don't help the investigation - the error message is sufficient, isn't it? What is anyone going to do with those bytes anyway?

Reply:
You would know who is the offending identity!

Reply:
identity %s for MSP %s is what tells you who the offender is. The rest is useless, isn't it?

Reply:
OK I removed the prints you don't like